From b93b61260b8eda33c01cef542777efbae2e8c570 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20H=C3=A4nel?= <florian.haenel@basyskom.com>
Date: Wed, 5 Jun 2013 19:23:32 +0200
Subject: [PATCH 1/2] Fix wrong calculation of viewPort for transitions

Viewport is calculated wrong for horizontal layout in a second instance

Task-number: QTBUG-29944
Task-Number: QTBUG-31546

Upstream-Status: Pending

Looks like backport of https://codereview.qt-project.org/49226 but this
one was applied in previous patch, this change looks the same but is
in different part of qquickitemview.cpp and ISN'T applied yet in dev
branch.

Signed-off-by: Florian Haenel <florian.haenel@lge.com>
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

Change-Id: I136ae4484278d14f796613505e3f38c65ec51ebe
---
 src/quick/items/qquickitemview.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/quick/items/qquickitemview.cpp b/src/quick/items/qquickitemview.cpp
index d774091..aa489eb 100644
--- a/src/quick/items/qquickitemview.cpp
+++ b/src/quick/items/qquickitemview.cpp
@@ -1815,7 +1815,7 @@ void QQuickItemViewPrivate::layout()
 
         prepareVisibleItemTransitions();
 
-        QRectF viewBounds(0, position(), q->width(), q->height());
+        QRectF viewBounds(q->contentX(),  q->contentY(), q->width(), q->height());
         for (QList<FxViewItem*>::Iterator it = releasePendingTransition.begin();
              it != releasePendingTransition.end(); ) {
             FxViewItem *item = *it;
-- 
1.8.2.1

